### PR TITLE
Graduate ElasticIndexedJob to GA

### PR DESCRIPTION
**Pull request by ahg-g**
Tracking issue: https://github.com/kubernetes/enhancements/issues/3715

### CODE REVIEW

**Comment by Shubham82** [comment]
Holding this PR for now, as the corresponding upstream PR hasn’t been merged yet.
[kubernetes/kubernetes#125751](https://github.com/kubernetes/kubernetes/pull/125751)

/hold
(remove the hold label once the upstream changes are merged)

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 1]
```diff
@@ -1004,13 +1004,10 @@ observe that pods from a Job are stuck with the tracking finalizer.
 
 ### Elastic Indexed Jobs
 
-{{< feature-state for_k8s_version="v1.27" state="beta" >}}
+{{< feature-state feature_gate_name="ElasticIndexedJob" >}}
```
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:1007]:
```suggestion
{{< feature-state for_k8s_version="v1.31" state="stable" feature_gate_name="ElasticIndexedJob" >}}
```

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 2]
```diff
@@ -1004,13 +1004,10 @@ observe that pods from a Job are stuck with the tracking finalizer.
...
 
 You can scale Indexed Jobs up or down by mutating both `.spec.parallelism` 
-and `.spec.completions` together such that `.spec.parallelism == .spec.completions`. 
-When the `ElasticIndexedJob`[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
-on the [API server](/docs/reference/command-line-tools-reference/kube-apiserver/)
-is disabled, `.spec.completions` is immutable.
+and `.spec.completions` together such that `.spec.parallelism == .spec.completions`.
```
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:1010]: Worth adding that, when scaling down, Kubernetes removes the pods with the higher indexes.

File: content/en/docs/concepts/workloads/controllers/job.md (hunk 1 above)
Comment (dipesh-rawat) [reply content/en/docs/concepts/workloads/controllers/job.md:1007]: The existing content `{{< feature-state feature_gate_name="ElasticIndexedJob" >}}` is accurate. We can check the [Build Preview link (here)](https://deploy-preview-46991--kubernetes-io-main-staging.netlify.app/docs/concepts/workloads/controllers/job/#elastic-indexed-jobs) to see how it renders. 

For information on using the `feature-state` shortcode, refer to the [documentation (here)](https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#feature-state-retrieval-from-description-file).

Comment (alculquicondor) [reply content/en/docs/concepts/workloads/controllers/job.md:1007]: Oh I see, so it pulls the information from the other file. Neat :)

File: content/en/docs/concepts/workloads/controllers/job.md (hunk 2 above)
Comment (ahg-g) [reply content/en/docs/concepts/workloads/controllers/job.md]: thanks, added

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 3]
```diff
@@ -1004,13 +1004,11 @@ observe that pods from a Job are stuck with the tracking finalizer.
...
 
-{{< feature-state for_k8s_version="v1.27" state="beta" >}}
+{{< feature-state feature_gate_name="ElasticIndexedJob" >}}
 
 You can scale Indexed Jobs up or down by mutating both `.spec.parallelism` 
 and `.spec.completions` together such that `.spec.parallelism == .spec.completions`. 
-When the `ElasticIndexedJob`[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
-on the [API server](/docs/reference/command-line-tools-reference/kube-apiserver/)
-is disabled, `.spec.completions` is immutable.
+When scaling down the higher indexes are removed.
```
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:1011]:
```suggestion
When scaling down, Kubernetes removes the Pods with the higher indexes.
```

active voice :)

Comment (ahg-g) [reply content/en/docs/concepts/workloads/controllers/job.md]: done

**Comment by katcosgrove** [comment]
/approve

Unhold when ready!

**Review by dipesh-rawat — COMMENTED** [review]
This doc PR looks good to _unhold_ now that the implementation for promoting the `ElasticIndexedJob` to GA has been merged into k/k as part of [PR-125751](https://github.com/kubernetes/kubernetes/pull/125751)

/unhold

**Pull request merged**